tests/generic/mongocursor-dead-001.phpt

```diff
@@ -25,7 +25,7 @@ $info = $txlogs->info();
 var_dump((string) $info['id']);
 ?>
 --EXPECTF--
-Not Dead
-Not Dead
+Dead
+Dead
```

Comment:
I believe it's correct because we set a limit of 2 on the MongoCursor before iterating and the entire collection has only two documents. Once iteration begins, the first (and only) batch should return both documents, `OP_REPLY` shouldn't specify a cursor ID for successive get more ops, and the cursor can immediately be considered dead.

I believe the only reason the cursor wasn't dead was because we were leaking data between test cases. I fixed this in https://github.com/jmikola/mongo-php-driver/commit/0e9d3713b3dfa25f0f27528fb19b0083959e393e.

Additionally, I added a variation of this test in mongocursor-dead-002.phpt, which performs the same limit=2 query, but on a collection with three documents. In that test, we assert that the cursor is not dead until we issue the one and only get more to fetch the third document.